web/docs/language/features.md

```diff
@@ -935,6 +939,82 @@ import AuthError from '@wasp/core/AuthError.js'
   }
 ```
 
+### Google
+
+`google` authentication makes it possible to use Google's OAuth 2.0 service to sign Google users into your app. To enable it, add `google: {}` to your `auth.methods` dictionary to use it with default settings. If you require custom configuration setup or a sign-in function, you can override the defaults.
+
+#### Default settings
+- Configuration:
+  - By default, we expect you to set two environment variables in order to use Google authentication:
+    - `GOOGLE_CLIENT_ID`
+    - `GOOGLE_CLIENT_SECRET`
+  - These can be obtained in your Google Cloud Console project dashboard. See [here](/docs/integrations/google#google-auth) for more.
+
+- Sign in:
+  - By default, when a user signs in with Google we will attempt to look up the user by email.
+    - If they exist, we will do nothing and return the JWT.
+    - If they do not exist, we will create a new user with a random password and return the JWT.
+
+- Here is a link to the default implementations: https://github.com/wasp-lang/wasp/blob/main/waspc/data/Generator/templates/server/src/routes/auth/passport/google/googleDefaults.js
+
+#### Overrides
+If you require modifications to the above, you can add one or more of the following to your `auth.methods.google` dictionary:
+
+```js
+  auth: {
+    userEntity: User,
+    methods: {
+      google: {
+        configFn: import { config } from "@ext/auth/google.js",
+        onSignInFn: import { signInHandler } from "@ext/auth/google.js"
+      }
+    },
+    ...
+  }
+```
+
+- `configFn`: This function should return an object with the following shape:
+```js
+export function config() {
+  return {
+    clientId: // look up from env or elsewhere,
+    clientSecret: // look up from env or elsewhere
+  }
+}
+```
+- `onSignInFn`: This function should return a user id:
+```js
+export async function signInHandler(context, args) {
+  // Find or create a context.entities.User based on args
+  return user
+}
+```
+
+#### UI helpers
+
+To use the Google sign-in button on your login page, simply do the following:
```

Comment:
What if I want to use my own button? Is there a method we expose for triggering the sign in, or smth else that is lower-level?